orbit/pkg/table/mdm/mdm_windows.go

```diff
@@ -0,0 +1,402 @@
+		outputCmd, err := executeMDMcommand(inputCmd)
+		if err != nil {
+			return nil, fmt.Errorf("mdm command execution: %s ", err)
+		}
+
+		log.Debug().Msgf("mdm_bridge output command response:\n%s", outputCmd)
+
+		return []map[string]string{
+			{
+				"enrollment_status":  "enrolled_device",
+				"enrolled_user":      enrollmentURI,
+				"mdm_command_input":  inputCmd,
+				"mdm_command_output": outputCmd,
+			},
+		}, nil
+	} else {
+		return []map[string]string{
+			{
+				"enrollment_status": "enrolled_device",
+				"enrolled_user":     enrollmentURI,
+				"mdm_command_input": inputCmd,
+				"mdm_command_ouput": "",
+			},
+		}, nil
+	}
+}
+
+// builtin utf16tostring string expects a uint16 array but we have a pointer to a uint16
+// so we need to cast it after converting it to an unsafe pointer
+// this is a common pattern though the buffer size varies
+// see https://golang.org/pkg/unsafe/#Pointer for more details
+func localUTF16toString(ptr unsafe.Pointer) (string, error) {
+	if ptr == nil {
+		return "", errors.New("failed UTF16 conversion due to null pointer")
+	} else {
+		uint16ptrarr := (*[maxBufSize]uint16)(ptr)[:]
+		return windows.UTF16ToString(uint16ptrarr), nil
+	}
+}
+
+// getEnrollmentInfo returns the MDM enrollment status by calling into OS API IsDeviceRegisteredWithManagement()
+func getEnrollmentInfo() (uint32, string, error) {
+	// variable to hold the MDM enrollment status
+	var isDeviceRegisteredWithMDM uint32 = 0
+
+	// heap-allocated buffer to hold the URI data
+	buffUriData := make([]uint16, 0, maxBufSize)
+	if buffUriData == nil {
+		return 0, "", errors.New("failed to allocate memory for URI data")
+	}
+
+	// IsDeviceRegisteredWithManagement is going to return the MDM enrollment status
+	// https://learn.microsoft.com/en-us/windows/win32/api/mdmregistration/nf-mdmregistration-isdeviceregisteredwithmanagement
+	if returnCode, _, err := procIsDeviceRegisteredWithManagement.Call(uintptr(unsafe.Pointer(&isDeviceRegisteredWithMDM)), maxBufSize, uintptr(unsafe.Pointer(&buffUriData))); returnCode != uintptr(windows.ERROR_SUCCESS) {
+		return 0, "", fmt.Errorf("there was an error calling IsDeviceRegisteredWithManagement(): %s (0x%X)", err, returnCode)
+	} else {
+
+		uriData, err := localUTF16toString(unsafe.Pointer(&buffUriData))
+		if err != nil {
+			return 0, "", err
+		}
+
+		return isDeviceRegisteredWithMDM, uriData, nil
+	}
+}
+
+// executeMDMcommand executes syncML MDM commands against the OS MDM stack and returns the status of the command execution
+func executeMDMcommand(inputCMD string) (string, error) {
+	// Synchronizing MDM command execution
+	mu.Lock()
+	defer mu.Unlock()
+
+	// Enabling MDM command executions
+	if err := enableCmdExecution(); err != nil {
+		return "", err
+	}
+
+	// Close MDM management mutex if neeeded - this is a hack to enable multiple MDM management calls
+	handle, err := windows.OpenMutex(windows.MUTEX_ALL_ACCESS, false, windows.StringToUTF16Ptr(mdmMutexName))
+	if err == nil {
```

Comment:
Not sure I understand that section, assuming it is required and working as intended :D

Reply:
The logic here was a hack to allow multiple executions of the MDM commands. The thing is that `ApplyLocalManagementSyncML()` opens up a mutex (named sync object) and does not close it, leaving the handle to that mutex opened, which keeps the mutex alive. Then, on a subsequent call to `ApplyLocalManagementSyncML()`, its internal logic (AcquireEnrollmentAccessMutex()) checks this mutex and fails to execute the command if the mutex is present. I found this after reverse engineering this undocumented function. So the hack here uses syscalls to traverse the list of handles that belongs to the current process and close the named mutex handle that was causing the issue. 

![image](https://user-images.githubusercontent.com/17311749/215165029-99113910-673d-4de2-afaf-6fc61413c010.png)